### DISCUSSION TITLE
Is there a way to pass arguments to the add action on repeater field?

### DISCUSSION THREAD

**Discussion by scottzirkel** [filamentphp/filament#10751] [Help] [answered]
I have a repeater with a custom view. In that view, I have a custom add action in which I'd like to pass through additional arguments from the front end.

My repeater field is named `elements`, so I'm passing the following through to the `data.elements` component. It's an add action with a simple payload.

Right now, I am able to add to the repeater, no problem. However, no matter where I look, I cannot find the arguments. I expected they would be under `$action->getArguments()` but that is always an empty array. My plan is to use these arguments as the default state of the new repeater block.

This button is in a loop of `$blocks`, so the `$block` value is populated from that.

```
<button
    type="button"
        wire:click="mountFormComponentAction('data.elements', 'add', {
        name: @js($block->name),
        data: @js($block->defaults),
    })"
        class="block"
>
    {{ $block->label }}
</button>
```

The `mountFormComponentAction` method clearly takes an array of arguments, so I'm fairly confused.

**Comment by webplusmultimedia** [comment, marked as answer]
Hi, 
You need to pass $arguments in any closure methods of the action (->form, ->action , etc) to retrieve them
```php
Action::make('name')
       ->action(static function (array $arguments, array $data, Action $action): array {
                    $name = $arguments['name'];
                    ....               

                    return [...];
         })
```

**Reply by scottzirkel** [reply]
Exactly what I was trying to accomplish. Thanks @webplusmultimedia!
